flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java

```diff
@@ -619,20 +619,20 @@ public void collect(StreamRecord<T> record) {
 		}
 
 		@Override
-		public <X> void collect(OutputTag<?> outputTag, StreamRecord<X> record) {
+		public <X> void collect(OutputTag<X> outputTag, StreamRecord<X> record) {
 			for (int i = 0; i < outputs.length - 1; i++) {
 				Output<StreamRecord<T>> output = outputs[i];
 
 				// due to side outputs, StreamRecords of varying types can pass through the broadcasting
 				// collector so we need to cast
 				@SuppressWarnings({"unchecked", "rawtypes"})
```

Comment:
Suppression not needed anymore. Also the comment is not adequate, as we do not cast anything.